pints/_mcmc/_slice_stepout.py

```diff
@@ -0,0 +1,430 @@
+        self._set_l = False
+        self._set_r  = False
+
+        # Index of parameter "xi" we are updating of the sample "x = (x1,...,xn)"
+        self._i = 0
+
+        # Stochastic variables as instance variables for testing
+        self._u = 0
+        self._v = 0
+        self._e = 0
+
+        # Flags used to calculate log_pdf of initial interval edges ``l,r```
+        self._init_left = False
+        self._init_right = False
+
+
+    def ask(self):
+        """ See :meth:`SingleChainMCMC.ask()`. """
+
+        # Check ask/tell pattern
+        if self._ready_for_tell:
+            raise RuntimeError('Ask() called when expecting call to tell().')
+
+        # Initialise on first call
+        if not self._running:
+            self._running = True
+
+        # Very first iteration
+        if self._current is None:
+
+            # Ask for the log pdf of x0
+            self._ready_for_tell = True
+            return np.array(self._x0, copy=True)
+
+
+        # If the flag is True, we initialise the expansion of interval ``I=(l,r)``
+        if self._first_expansion == True:
+
+            # Set initial values for l and r
+            self._u = np.random.uniform()
+            self._l = self._proposed[self._i] - self._w[self._i]*self._u
+            self._r = self._l + self._w[self._i]
+
+            # Set maximum number of steps for expansion to the left (j) and right (k)
+            self._v = np.random.uniform()
+            self._j = np.floor(self._m*self._v)
+            self._k = (self._m-1) - self._j
+
+            # Initialise arrays used for calculating the log_pdf of the edges l,r
+            self._temp_l = np.array(self._proposed, copy=True)
+            self._temp_r = np.array(self._proposed, copy=True)
+            self._temp_l[self._i] = self._l
+            self._temp_r[self._i] = self._r
+
+            # We have initialised the expansion, so we set the flag to false
+            self._first_expansion = False
+
+            # Set flags to calculate log_pdf of ``l,r``
+            self._init_left = True
+            self._init_right = True
+
+        # Ask for log_pdf of initial edges ``l,r```
+        if self._init_left:
+
+            # Ask for log_pdf of initial left edge
+            self._ready_for_tell = True
+            return np.array(self._temp_l, copy=True)
+
+        if self._init_right:
+
+            # Ask for log pdf of initial right edge
+            self._ready_for_tell = True
+            return np.array(self._temp_r, copy=True)
+
+        # Expand the interval ``I``` until edges ``l,r`` are outside the slice or we have reached
+        # limit of expansion steps
+
+        # Check whether we can expand to the left
+        if self._j > 0 and self._current_log_y < self._fx_l:
+
+            # Set flag to indicate that we are updating the left edge
+            self._set_l = True
+
+            # If left edge of the interval is inside the slice, keep expanding
```

Comment:
There's no if at this point so we just expand. So think we should comment differently.